File: rmats-viz/frontend/src/types/event.ts
export type EventType = "SE" | "RI" | "A3SS" | "A5SS" | "MXE";

export interface SplicingEvent {
  id: string;
  analysis_id: string;
  event_type: EventType;
  rmats_id?: number | null;
  gene_id?: string | null;
  gene_symbol?: string | null;
  chr?: string | null;
  strand?: string | null;
  exon_start?: number | null;
  exon_end?: number | null;
  upstream_es?: number | null;
  upstream_ee?: number | null;
  downstream_es?: number | null;
  downstream_ee?: number | null;
  second_exon_start?: number | null;
  second_exon_end?: number | null;
  ijc_sample_1?: string | null;
  sjc_sample_1?: string | null;
  ijc_sample_2?: string | null;
  sjc_sample_2?: string | null;
  p_value?: number | null;
  fdr?: number | null;
  inc_level_1?: string | null;
  inc_level_2?: string | null;
  inc_level_difference?: number | null;
  abs_inc_level_diff?: number | null;
  top_rank?: number | null;
}

export interface EventsPage {
  items: SplicingEvent[];
  total: number;
  page: number;
  page_size: number;
  pages: number;
}

File: rmats-viz/backend/app/schemas/event.py
from __future__ import annotations
import uuid
from pydantic import BaseModel


class SplicingEventResponse(BaseModel):
    id: uuid.UUID
    analysis_id: uuid.UUID
    event_type: str
    rmats_id: int | None = None
    gene_id: str | None = None
    gene_symbol: str | None = None
    chr: str | None = None
    strand: str | None = None
    exon_start: int | None = None
    exon_end: int | None = None
    upstream_es: int | None = None
    upstream_ee: int | None = None
    downstream_es: int | None = None
    downstream_ee: int | None = None
    second_exon_start: int | None = None
    second_exon_end: int | None = None
    ijc_sample_1: str | None = None
    sjc_sample_1: str | None = None
    ijc_sample_2: str | None = None
    sjc_sample_2: str | None = None
    p_value: float | None = None
    fdr: float | None = None
    inc_level_1: str | None = None
    inc_level_2: str | None = None
    inc_level_difference: float | None = None
    abs_inc_level_diff: float | None = None
    top_rank: int | None = None

    model_config = {"from_attributes": True}


class EventsPage(BaseModel):
    items: list[SplicingEventResponse]
    total: int
    page: int
    page_size: int
    pages: int
